upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go

```diff
@@ -134,7 +135,12 @@ func (e *Elastigroup) Find(c *fi.Context) (*Elastigroup, error) {
 		{
 			for _, zone := range compute.AvailabilityZones {
 				if zone.SubnetID != nil {
-					actual.Subnets = append(actual.Subnets, &awstasks.Subnet{ID: zone.SubnetID})
+					t := &awstasks.Subnet{ID: zone.SubnetID}
+					s, err := t.Find(c)
```

Comment:
This change _shouldn't_ be necessary - Subnet implements `CompareWithID` and returns `ID` as the comparison value https://github.com/kubernetes/kops/blob/master/upup/pkg/fi/cloudup/awstasks/subnet.go#L55 .  When we're comparing fields for changes, we should therefore consider two Subnets with the same ID the same, even if the other fields are different. 

But likely I just don't understand why you need to populate the other fields?